utils/generate_backend_completeness.py

```diff
@@ -0,0 +1,109 @@
+from __future__ import annotations
+
+import importlib
+import inspect
+from pathlib import Path
+from typing import Any
+from typing import Final
+
+import polars as pl
+from jinja2 import Template
+
+TEMPLATE_PATH: Final[Path] = Path("utils") / "api-completeness.md.jinja"
+DESTINATION_PATH: Final[Path] = Path("docs") / "api-reference" / "api-completeness.md"
+
+
+MODULES = ["dataframe", "series", "expression"]
+EXCLUDE_CLASSES = {"BaseFrame"}
+
+
+def get_class_methods(kls: type[Any]) -> list[str]:
+    return [m[0] for m in inspect.getmembers(kls) if not m[0].startswith("_")]
+
+
+def get_backend_completeness_table() -> str:
+    results = []
+
+    for module_name in MODULES:
+        nw_namespace = f"narwhals.{module_name}"
+        sub_module_name = module_name if module_name != "expression" else "expr"
+
+        narwhals_module_ = importlib.import_module(nw_namespace)
+        classes_ = inspect.getmembers(
+            narwhals_module_,
+            predicate=lambda c: inspect.isclass(c) and c.__module__ == nw_namespace,  # noqa: B023, not imported classes
+        )
+
+        for nw_class_name, nw_class in classes_:
+            if nw_class_name in EXCLUDE_CLASSES:
+                continue
```

Comment:
Skipping `BaseFrame` but more can be added easily

Reply:
we should probably remove `BaseFrame` completely tbh